content/en/docs/concepts/security/mutating-webhooks-good-practices.md

```diff
@@ -0,0 +1,585 @@
+
+When a node that runs the webhook server Pods becomes unhealthy, the webhook
+Deployment tries to reschedule the Pods to another node. However, the existing
+webhook server rejects the requests since the `"env"` label is unset. As a
+result, the migration cannot happen.
+
+Exclude the namespace where your webhook is running with a
+[namespaceSelector](/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-namespaceselector).
+
+### Fail open and validate the final state {#fail-open-validate-final-state}
+
+Mutating admission webhooks support the `failurePolicy` configuration field.
+This field indicates whether the API server should admit or reject the request
+if the webhook fails. Webhook failures might occur because of timeouts or errors
+in the server logic.
+
+By default, admission webhooks set the `failurePolicy` field to Fail. The API
+server rejects a request if the webhook fails. However, rejecting requests by
+default might result in compliant requests being rejected during webhook
+downtime. 
+
+Let your mutating webhooks "fail open" by setting the `failurePolicy` field to
+Ignore. Use a validating controller to check the state of requests to ensure
+that they comply with your policies. 
+
+This approach has the following benefits:
+
+* Mutating webhook downtime doesn't affect compliant resources from deploying.
+* Policy enforcement occurs during validating admission control.
+* Mutating webhooks don't interfere with other controllers in the cluster.
+
+### Plan for future updates to fields {#plan-future-field-updates}
+
+In general, design your webhooks under the assumption that Kubernetes APIs might
+change in a later version. Don't write a server that takes the stability of an
+API for granted. For example, the release of sidecar containers in Kubernetes
+added a `restartPolicy` field to the Pod API. 
+
+### Prevent your webhook from triggering itself {#prevent-webhook-self-trigger}
+
+Mutating webhooks that respond to a broad range of API requests might
+unintentionally trigger themselves. For example, consider a webhook that
+responds to all requests in the cluster. If you configure the webhook to create
+`Event` objects for every mutation, it'll respond to its own `Event` object
+creation requests.
+
+To avoid this, consider setting a unique label in any resources that your
+webhook creates. Exclude this label from your webhook match conditions.
+
+### Don't change immutable objects {#dont-change-immutable-objects}
+
+Some Kubernetes objects in the API server can't change. For example, when you
+deploy a {{< glossary_tooltip text="static Pod" term_id="static-pod" >}}, the
+kubelet on the node creates a 
+{{< glossary_tooltip text="mirror Pod" term_id="mirror-pod" >}} in the API
+server to track the static Pod. However, changes to the mirror Pod don't
+propagate to the static Pod. 
+
+Don't attempt to mutate these objects during admission. 
```

Comment:
- I would mention that mirror pods have the `kubernetes.io/config.mirror` annotation.
- To avoid security problems, consider _also_ enforcing an allow list of namespaces that can contain mirror pods.

I don't know a better way to detect mirror Pods (eg: from their `.status` field)